CRM/Utils/Check/Component/Timestamps.php

```diff
@@ -18,6 +18,48 @@ class CRM_Utils_Check_Component_Timestamps extends CRM_Utils_Check_Component {
 
   const DOCTOR_WHEN = 'https://github.com/civicrm/org.civicrm.doctorwhen';
 
+  /**
+   * Check that MySQL actually supports timezone operations.
+   *
+   * @return CRM_Utils_Check_Message[]
+   */
+  public function checkTimezoneAPIs() {
+    $messages = [];
+
+    try {
+      $tzCount = CRM_Core_DAO::singleValueQuery('SELECT count(*) FROM mysql.time_zone_name');
+    }
+    catch (\Exception $e) {
+      $tzCount = 0;
+    }
+
+    try {
+      $convertedTime = CRM_Core_DAO::singleValueQuery('SELECT CONVERT_TZ("2001-02-03 04:05:00", "GMT", "America/New_York")');
+    }
+    catch (\Exception $e) {
+      $convertedTime = NULL;
+    }
+    $expectedTime = '2001-02-02 23:05:00';
+
+    if ($tzCount < 5 || $convertedTime !== $expectedTime) {
```

Comment:
If I run this on my local where I don't have timezones, I get the warning as expected. But I also ran it on a system that has timezones and I still got the warning, but it's because this is an OR clause, and the user doesn't have access to the mysql.* tables, which is probably common.

Reply:
@demeritcowboy I had originally just thought to do the  `mysql.time_zone_name` check. But we could flip that around. Do you think we could get away with just testing `CONVERT_TZ()`?

My only hesitation is a gut fear that "a broken clock is right, twice a day" (*maybe `CONVERT_TZ()` passes or fails because of some coincidence in how the test-values resemble the local-configuration*). It's probably misplaced here -- I can't see a concrete way that the `CONVERT_TZ()` test would go awry.

Another option: send two `CONVERT_TZ()` calls (with 4 distinct timezones at play). That would pretty squarely rule-out the possibility of a timezone coincidence.

Reply:
Could do two CONVERT_TZ just to be sure. Could also do something like `$oldzone = SELECT @@time_zone;` followed by `SET time_zone = 'America/New_York'` and then read it back and see if it updated, then replace old zone. But I suppose you'd have to do that twice or more too to be more sure it has a full range of zones loaded.

Reply:
@demeritcowboy OK, I updated to use the `SET time_zone` trick (and remove the `mysql.time_zone_name` trick). So it should be a little more forgiving about system permissions.